scripts/deterministic/DeployScroll.s.sol

```diff
@@ -186,6 +196,14 @@ contract DeployScroll is DeterminsticDeployment {
         _;
     }
 
+    /// @dev Only execute block if it's requied by alternative gas token mode.
+    modifier gasToken(bool gasTokenRequire) {
+        if (ALTERNATIVE_GAS_TOKEN_ENABLED != gasTokenRequire) {
```

Comment:
weird, IMO, if set `ALTERNATIVE_GAS_TOKEN_ENABLED=true` represents `alternative gas token` enabled, why need pass `gasTokenRequire` to check the consistency. if `ALTERNATIVE_GAS_TOKEN_ENABLED=fase && gasTokenRequire == false`, this check `(ALTERNATIVE_GAS_TOKEN_ENABLED != gasTokenRequire)` will also verify pass.

Reply:
Some functions are only run when alternative gas token is enabled: `gasToken(true)`. Others are run when alternative gas token is disabled: `gasToken(false)`.

You need both, `ALTERNATIVE_GAS_TOKEN_ENABLED` is the user config for the rollup, while `gasTokenRequire` is for the script to decide to enable/disable certain functions.

If both are false then the check will evaluate to false (`false != false` is false), i.e. we will run the function instead of returning early.